redis/tests/support/mod.rs

```diff
@@ -286,15 +307,71 @@ pub struct TestContext {
     pub client: redis::Client,
 }
 
+pub(crate) fn is_tls_enabled() -> bool {
```

Comment:
I think it might be a simpler approach to just have a few very basic tests (i.e., for sync, async, clustered, etc.) that verify the new functionality works. I would prefer we avoid enabling it for all tls-enabled tests.